std/experimental/allocator/building_blocks/allocator_list.d

```diff
@@ -632,119 +637,236 @@ version(Posix) @system unittest
     assert(b1.length == 1024 * 10);
 }
 
+/// Ditto
+shared struct SharedAllocatorList(Factory, BookkeepingAllocator = GCAllocator)
+{
+    import std.typecons : Ternary;
+    import core.internal.spinlock : SpinLock;
+    import std.traits : hasMember, isCallable, Parameters;
+    import std.algorithm.searching : startsWith;
+
+private:
+    // Forward all calls to 'impl' and protect them by the lock below
+    AllocatorList!(Factory, BookkeepingAllocator) impl;
+    SpinLock lock = SpinLock(SpinLock.Contention.brief);
+
+    // Remove this when 'https://github.com/dlang/druntime/pull/2156' gets merged
+    static ref T assumeUnshared(T)(ref shared T val) @trusted @nogc pure nothrow
+    {
+        return *cast(T*) &val;
+    }
+
+// Everything is inherited from the 'AllocatorList' implementation
+public:
+    /**
+    The alignment offered.
+    */
+    enum alignment = impl.alignment;
+
+    static foreach (T; __traits(allMembers, typeof(impl)))
```

Comment:
Please add a comment before this loop explaining what it does. Cool idea @wilzbach !!

Reply:
AFAICT this loop got removed. Nothing left to comment.